internal/services/storagecache/managed_lustre_file_system_resource.go

```diff
@@ -494,6 +537,42 @@ func expandManagedLustreFileSystemEncryptionKey(input []EncryptionKey) *amlfiles
 	}
 }
 
+func expandRootSquashSettings(input []RootSquashSetting) *amlfilesystems.AmlFilesystemRootSquashSettings {
+	if len(input) == 0 {
+		return nil
+	}
+
+	rootSquashSetting := &input[0]
+
+	result := &amlfilesystems.AmlFilesystemRootSquashSettings{
+		Mode:             pointer.To(amlfilesystems.AmlFilesystemSquashMode(rootSquashSetting.Mode)),
+		NoSquashNidLists: pointer.To(rootSquashSetting.NoSquashNidList),
+		SquashGID:        pointer.To(rootSquashSetting.SquashGID),
+		SquashUID:        pointer.To(rootSquashSetting.SquashUID),
+	}
+
+	return result
```

Comment:
Currently when this returns `nil` if `root_squash_setting` is no longer present in the config, the update wouldn't actually remove the root squash configuration. If `len(input) == 0` I think we can return `mode: "None"` to ensure it's always removed unless specified.

Additionally we can also just return the result right away instead of first assigning it to a variable

```suggestion
	if len(input) == 0 {
		return &amlfilesystems.AmlFilesystemRootSquashSettings{
			Mode: pointer.To(amlfilesystems.AmlFilesystemSquashModeNone),
		}
	}

	rootSquashSetting := &input[0]

	return &amlfilesystems.AmlFilesystemRootSquashSettings{
		Mode:             pointer.To(amlfilesystems.AmlFilesystemSquashMode(rootSquashSetting.Mode)),
		NoSquashNidLists: pointer.To(rootSquashSetting.NoSquashNidList),
		SquashGID:        pointer.To(rootSquashSetting.SquashGID),
		SquashUID:        pointer.To(rootSquashSetting.SquashUID),
	}
```